File: fix_use_client_directive.js
const fs = require('fs');
const path = require('path');

// Function to fix "use client" directive placement
function fixUseClientDirective(content) {
  const lines = content.split('\n');
  let useClientLine = -1;
  let firstImportLine = -1;
  
  // Find "use client" line and first import line
  for (let i = 0; i < lines.length; i++) {
    const line = lines[i].trim();
    if (line === "'use client'" || line === '"use client"') {
      useClientLine = i;
    }
    if (line.startsWith('import ') && firstImportLine === -1) {
      firstImportLine = i;
    }
  }
  
  // If "use client" exists but is not at the top
  if (useClientLine > 0) {
    const useClientDirective = lines[useClientLine];
    
    // Remove the "use client" line from its current position
    lines.splice(useClientLine, 1);
    
    // Insert it at the very beginning
    lines.unshift(useClientDirective);
    
    return lines.join('\n');
  }
  
  return content;
}

// Function to process a single file
function processFile(filePath) {
  try {
    const content = fs.readFileSync(filePath, 'utf8');
    
    if (content.includes("'use client'") || content.includes('"use client"')) {
      console.log(`Processing: ${filePath}`);
      const fixed = fixUseClientDirective(content);
      
      if (fixed !== content) {
        fs.writeFileSync(filePath, fixed, 'utf8');
        return true;
      }
    }
    return false;
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

// Function to recursively find and process files
function processDirectory(dirPath) {
  let processedCount = 0;
  
  try {
    const items = fs.readdirSync(dirPath);
    
    for (const item of items) {
      const fullPath = path.join(dirPath, item);
      const stat = fs.statSync(fullPath);
      
      if (stat.isDirectory()) {
        if (!['node_modules', '.git', '.next', 'dist', 'build'].includes(item)) {
          processedCount += processDirectory(fullPath);
        }
      } else if (stat.isFile()) {
        if (/\.(tsx?|jsx?)$/.test(item)) {
          if (processFile(fullPath)) {
            processedCount++;
          }
        }
      }
    }
  } catch (error) {
    console.error(`Error processing directory ${dirPath}:`, error.message);
  }
  
  return processedCount;
}

// Main execution
console.log('Starting "use client" directive fix...');
const processedCount = processDirectory('./app');
console.log(`Processed ${processedCount} files with "use client" directive issues.`);

if (fs.existsSync('./src')) {
  const srcProcessedCount = processDirectory('./src');
  console.log(`Processed ${srcProcessedCount} files in src directory.`);
}

console.log('"use client" directive fix completed!');

File: fix_use_client_directive.py
#!/usr/bin/env python3
"""
Fix 'use client' directive placement in Next.js files.
This script ensures 'use client' is at the very top of files.
"""

import os
import re
import glob
from pathlib import Path

def fix_use_client_directive(file_path):
    """Fix 'use client' directive placement in a single file."""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Skip if no 'use client' directive
        if "'use client'" not in content and '"use client"' not in content:
            return False
        
        # Check if 'use client' is already at the top
        lines = content.split('\n')
        if lines[0].strip() == "'use client'" or lines[0].strip() == '"use client"':
            return False
        
        # Find and remove existing 'use client' directive
        content = re.sub(r"^['\"]use client['\"];\s*\n?", '', content, flags=re.MULTILINE)
        content = re.sub(r"\n\s*['\"]use client['\"];\s*\n?", '\n', content)
        
        # Add 'use client' at the very top
        content = "'use client';\n" + content
        
        # Write the fixed content back
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)
        
        print(f"Fixed 'use client' directive placement: {file_path}")
        return True
        
    except Exception as e:
        print(f"Error fixing 'use client' directive in {file_path}: {e}")
        return False

def find_tsx_files():
    """Find all TypeScript React files."""
    files = []
    
    # Search for TypeScript React files
    patterns = [
        'app/**/*.tsx',
        'src/**/*.tsx',
        'components/**/*.tsx'
    ]
    
    for pattern in patterns:
        for file_path in glob.glob(pattern, recursive=True):
            if os.path.isfile(file_path):
                files.append(file_path)
    
    return files

def main():
    """Main function to fix 'use client' directive placement."""
    print("Starting 'use client' directive fixes...")
    
    # Find all TSX files
    tsx_files = find_tsx_files()
    print(f"Found {len(tsx_files)} TSX files")
    
    fixed_count = 0
    
    for file_path in tsx_files:
        if fix_use_client_directive(file_path):
            fixed_count += 1
    
    print(f"\n'use client' directive fixes complete:")
    print(f"  - Fixed: {fixed_count} files")

if __name__ == "__main__":
    main()